src/components/account/details/UpdateToggle.js

```diff
@@ -1,13 +1,46 @@
 import React, { Component } from 'react'
+import { connect } from 'react-redux'
+
+import { toggleEdit } from '../../../actions/account_details'
+import Button from './Button'
 
 import '../../../styles/account/details/update_toggle.css'
 
-export default class UpdateToggle extends Component {
+class UpdateToggle extends Component {
+  handleClick() {
+    this.props.toggleEdit()
+  }
+
   render() {
-    return (
-      <div className="edit-details__button">
-        <span>{ this.props.name }</span>
-      </div>
-    )
+    if ( !this.props.editing ) {
```

Comment:
Also, this looks like it might be a great opportunity for sub components - there's a lot of rendering going on here, and it looks like the things that are getting rendered are very distinct in terms of their behavior and functionality.